### PR TITLE
Simplify definition and usage of TestDirectory test rule.

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Remove outer TargetDirectory class, let users avoid specifying class under test.

With this PR typical definition of TestDirectory look like:

```
@Rule
public TestDirectory testDirectory = TestDirectory.testDirectory();
```

### CODE REVIEW

**Comment by chrisvest** [comment]
Without giving the test classes, how do you find the specific store for a test that fails?

**Comment by chrisvest** [comment]
Oh, I see. The rules infrastructure can give it to you.

**Comment by chrisvest** [comment]
@MishaDemianenko Needs a rebase

File: community/io/src/test/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCacheWithRealFileSystemIT.java [hunk 1]
```diff
@@ -19,15 +19,24 @@
...
 {
     @Rule
-    public TargetDirectory.TestDirectory directory = TargetDirectory.testDirForTest( getClass() );
+    public TestDirectory directory = TestDirectory.testDirectory();
+
+    @After
+    public void tearDown() throws IOException
+    {
+        directory.cleanup();
+    }
```
Comment (chrisvest) [inline community/io/src/test/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCacheWithRealFileSystemIT.java]: Why are these cleanups needed? The `tearDown` method is called regardless of the outcome of the test, but we generally desire to keep the test data around if the test fails. Each test should also be working with their own dedicated directory, so there should be no risk of test isolation failures.

Comment (MishaDemianenko) [inline community/io/src/test/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCacheWithRealFileSystemIT.java]: You absolutely right, it's leftovers; used page cache tests as one of the first passing criteria:)

**Comment by MishaDemianenko** [comment]
@chrisvest rebased and cleaned

**Pull request merged**